lib/github-interface.js

```diff
@@ -0,0 +1,232 @@
+//   repos: ['tableflip/project-frontend', 'tableflip/project-api'],
+//   version: 'v2.0.0',
+//   message: 'Adds notifications and basic admin console'
+// })
+async function prerelease ({ repos, version, message }) {
+  const tags = repos.map((repo) => tagRepo({ repo, version }))
+  await Promise.all(tags)
+  return prereleaseRepo({ repo: repos[0], version, message })
+}
+
+// Adds a the supplied version tag to the current master branch in the given repo
+async function tagRepo ({ repo, version }) {
+  const lastCommit = await request({
+    method: 'GET',
+    url: `${githubApiBase}/repos/${repo}/commits/master`,
+    auth,
+    json: true,
+    headers: githubHeaders
+  })
+
+  if (!lastCommit || !lastCommit.sha) throw new Error(`Cannot get latest sha for the repo ${repo}`)
+
+  return request({
+    method: 'POST',
+    url: `${githubApiBase}/repos/${repo}/git/refs`,
+    auth,
+    json: {
+      ref: `refs/tags/${version}`,
+      sha: lastCommit.sha
+    },
+    headers: githubHeaders
+  })
+}
+
+// Adds a release (with the prerelease flag) to the supplied version, with the supplied message, to the given repo
+function prereleaseRepo ({ repo, version, message }) {
+  return request({
+    method: 'POST',
+    url: `${githubApiBase}/repos/${repo}/releases`,
+    auth,
+    json: {
+      tag_name: version,
+      target_commitish: 'master',
+      name: version,
+      body: message,
+      prerelease: true
+    },
+    headers: githubHeaders
+  })
+}
+
+// Gets the package.json for a repo, assuming it's located in the root.  Throws if no such package.json is found
+async function getPackageJson ({ repo }) {
+  const res = await request({
+    method: 'GET',
+    url: `${githubApiBase}/repos/${repo}/contents/package.json`,
+    auth,
+    json: true,
+    headers: githubHeaders
+  })
+  const buf = Buffer.from(res.content, 'base64')
+  return {
+    json: JSON.parse(buf.toString('utf8')),
+    sha: res.sha
+  }
+}
+
+// Returns the version number from the given repo's package.json, or throws if it can't be found
+async function getVersion ({ repo }) {
+  const { json } = await getPackageJson({ repo })
+  return json.version
+}
+
+// Returns the supplied version number bumped for the next release.
+// Default releaseType is a 'patch', but could be 'major' or 'minor'.
+function bumpVersion ({ version, releaseType = 'patch' }) {
+  return semver.inc(version, releaseType)
+}
+
+// Sets the version number in the given repo to the supplied number, assuming it's in the repo root.
+// Throws if the repo or package.json can't be found.
+async function setVersion ({ repo, version }) {
+  const { json, sha } = await getPackageJson({ repo })
+  json.version = version
+  const content = Buffer.from(JSON.stringify(json, null, 2), 'utf8').toString('base64')
+  return request({
+    method: 'PUT',
+    url: `${githubApiBase}/repos/${repo}/contents/package.json`,
+    auth,
+    json: {
+      path: 'package.json',
+      message: 'Updates version',
```

Comment:
`npm version` uses the version number as the commit message (without "v" prefix) so it would be good to have parity with that.